blenderproc/python/loader/URDFLoader.py

```diff
@@ -0,0 +1,270 @@
+                               depth=geometry_tree.cylinder.length)
+    elif geometry_tree.sphere is not None:
+        obj = create_primitive(shape="SPHERE", radius=geometry_tree.sphere.radius)
+    else:
+        raise NotImplementedError(f"Unknown geometry in urdf_tree {geometry_tree}")
+    bpy.ops.object.transform_apply(location=False, rotation=False, scale=True, properties=True)
+    return obj
+
+
+def load_viscol(viscol_tree: Union["urdfpy.Visual", "urdfpy.Collision"], name: str) -> MeshObject:
+    """ Loads a visual / collision element from an urdf tree.
+
+    :param viscol_tree: The urdf representation of the visual / collision element.
+    :return: The respective MeshObject.
+    """
+    obj = load_geometry(viscol_tree.geometry)
+    try:
+        obj.set_name(name=viscol_tree.name)
+    except Exception as e:
+        # fallback to the link name with specified suffix
+        print(f"Didn't find a name for visual / collision object in {viscol_tree} ({e}). Falling back to {name}.")
+        obj.set_name(name=name)
+
+    # load material - only valid for visuals
+    if hasattr(viscol_tree, "material"):
+        # clear all existing materials
+        obj.clear_materials()
+
+        # check if material exists
+        if viscol_tree.material.name in [m.name for m in get_all_materials()]:
+            mat = bpy.data.materials[viscol_tree.material.name]
+            mat = Material(mat)
+        else:
+            # create a new material
+            mat = MaterialLoaderUtility.create(name=viscol_tree.material.name)
+            # create a principled node and set the default color
+            principled_node = mat.get_the_one_node_with_type("BsdfPrincipled")
+            principled_node.inputs["Base Color"].default_value = viscol_tree.material.color
+        obj.replace_materials(mat)
+
+        # check for textures
+        if viscol_tree.material.texture is not None:
+            # image should've been loaded automatically
+            mat = MaterialLoaderUtility.create(name=viscol_tree.material.name + "_texture")
+            nodes = mat.nodes
+            links = mat.links
+
+            color_image = nodes.new('ShaderNodeTexImage')
+
+            if not os.path.exists(viscol_tree.material.texture.filename):
+                raise Exception(f"Couldn't load texture image for {viscol_tree} from {viscol_tree.material.texture.filename}")
+            color_image.image = bpy.data.images.load(viscol_tree.material.texture.filename, check_existing=True)
+
+            principled = Utility.get_the_one_node_with_type(nodes, "BsdfPrincipled")
+            links.new(color_image.outputs["Color"], principled.inputs["Base Color"])
+
+            if obj.blender_obj.data.materials:
+                # assign to 1st material slot
+                obj.blender_obj.data.materials[0] = mat
+            else:
+                # no slots
+                obj.blender_obj.data.materials.append(mat)
```

Comment:
This should be equivalent to
```suggestion
            obj.replace_materials(mat)
```